Support case when plymouth starts from initramfs

alfonsosanchezbeato opened the pull request:
Support case when plymouth starts from initramfs, by using the
/run/plymouth/pid path as an indication of whether plymouthd is
running instead of the plymouth-start service. If plymouthd has not
been started by the initramfs the behavior will be as before.

alfonsosanchezbeato committed:
Support case when plymouth starts from initramfs

Support case when plymouth starts from initramfs, by using the
/run/plymouth/pid path as an indication of whether plymouthd is
running instead of the plymouth-start service. If plymouthd has not
been started by the initramfs the behavior will be as before.

stulluk reviewed (approved):
LGTM

mvo5 commented on config/lib/systemd/system/plymouth-show-run.service:
```diff
@@ -1,6 +1,5 @@
 [Unit]
 Description=Display state on splash
-After=plymouth-start.service
```
I wonder if it is really okay to remove this "After=" relation - in the cases where plymouth is not started from the initrd we still need the ordering that plymouth-start.service is started first and then the services that use it are run. Or am I missing something here? AIUI having the After even if the service is not started is fine as it's just a weak dependency

alfonsosanchezbeato replied:
The ordering should be guaranteed as `ConditionPathExists=/run/plymouth/pid` applies in both cases. But you are right that adding `After=` should be fine too. I've put that back.

alfonsosanchezbeato committed:
Put back After=plymouth-start condition for some units

As this is a weak condition anyway, it won't do any harm even when
plymouthd has started from the initramfs.

mvo5 reviewed (approved):
Thank you

alfonsosanchezbeato commented:
Thank you!
